### PR TITLE
Bug 1950133: Add cleanupStatus to keep three kubelet config conditions

### PR DESCRIPTION
**Pull request by QiWang19**
Signed-off-by: Qi Wang <qiwan@redhat.com>

Close: Bug 1950133 https://bugzilla.redhat.com/show_bug.cgi?id=1950133

**- What I did**
Add cleanUpStatus method to keep a limited number of conditions of different
timestamps. The list keeps the three latest conditions. Updates only the timestamp
if the new status condition stays the same as the last condition.

**- How to verify it**

**- Description for the changelog**

fix the bug MCO creates empty conditions on the kubeletconfig object

### CODE REVIEW

**Comment by QiWang19** [comment]
@umohnani8 @rphillips PTAL

**Comment by umohnani8** [comment]
LGTM
/retest

File: pkg/controller/kubelet-config/kubelet_config_controller.go [hunk 1]
```diff
@@ -346,16 +346,13 @@ func (ctrl *Controller) syncStatusOnly(cfg *mcfgv1.KubeletConfig, err error, arg
 		if getErr != nil {
 			return getErr
 		}
-		// To avoid a long list of same statuses, only append a status if it is the first status
+		// Keeps a list of three status to avoid a long list of same statuses,
```
Comment (kikisdeliveryservice) [inline pkg/controller/kubelet-config/kubelet_config_controller.go:349]: thanks for the clear comments here =)

**Review by kikisdeliveryservice — APPROVED** [review]
Thanks @QiWang19 Appreciate the tests as well !

**Pull request merged**